framework/test/unit/engine/consensus/certificate_generation/commit_pool.spec.ts

```diff
@@ -241,7 +235,91 @@ describe('CommitPool', () => {
 			expect(commitPool['_gossipedCommits'].getByHeight(1070)).toBeArrayOfSize(0);
 		});
 
-		it('should select non gossiped commits that are created by the generator of the engine', async () => {
+		it(`should not clean commits for the heights ${maxHeightPrecommitted} - ${COMMIT_RANGE_STORED} until currentHeight and bftParams does not exist for height`, async () => {
+			// // Update current height so that commits will always be in the future
+			chain.lastBlock = { header: { height: 1019 } };
+
+			// it should not be deleted by the height
+			commitPool['_nonGossipedCommits'].add({
+				blockID: utils.getRandomBytes(32),
+				certificateSignature: utils.getRandomBytes(96),
+				height: 900,
```

Comment:
I think it will be better if we can calculate this value based on `maxHeightPrecommitted` and `COMMIT_RANGE_STORED` so that we don't need to explain how we came up with this magic number. So `maxHeightPrecommitted - COMMIT_RANGE_STORED`